### PR TITLE
Fix/papulatus entry

### PR DESCRIPTION
**Pull request by allenaerostar**

### CODE REVIEW

File: src/net/server/world/World.java [hunk 1]
```diff
@@ -103,6 +106,7 @@ public World(int world, int flag, String eventmsg, int exprate, int droprate, in
         petsSchedule = TimerManager.getInstance().register(new PetFullnessWorker(this), 60 * 1000, 60 * 1000);
         mountsSchedule = TimerManager.getInstance().register(new MountTirednessWorker(this), 60 * 1000, 60 * 1000);
         charactersSchedule = TimerManager.getInstance().register(new CharacterAutosaverWorker(this), 60 * 60 * 1000, 60 * 60 * 1000);
+        bossEntriesSchedule = TimerManager.getInstance().register(new BossEntriesWorker(this), calculateDifferenceTillMidNight(), 60 * 1000 * 1440);
```
Comment (nansonzheng) [inline src/net/server/world/World.java]: Swap calculateDifferenceTillMidNight() and 60 * 1000 * 1440. As confusing as it is, src/server/TimerManager.java swaps the positions for delay and repeatTime from ScheduledThreadPoolExecutor.

**Review by nansonzheng — CHANGES_REQUESTED** [review]

File: src/client/MapleCharacter.java [hunk 2]
```diff
@@ -6002,8 +6022,26 @@ public synchronized void saveToDB(boolean notAutosave) {
             ps.setInt(1, gmLevel > 1 ? 1 : 0);
             ps.setInt(2, client.getAccID());
             ps.executeUpdate();
+
+            ps = con.prepareStatement("SELECT * FROM boss_entries WHERE charid = ?");
```
Comment (BenjixD) [inline src/client/MapleCharacter.java]: refrain from using select * whenever possible

File: src/net/server/world/World.java (hunk 1 above)
Comment (allenaerostar) [reply src/net/server/world/World.java]: Oh noted!

File: src/net/server/world/World.java [hunk 3]
```diff
@@ -774,6 +778,37 @@ public void runMountSchedule() {
         }
     }
 
+    //reset all characters boss_entries to world defined maximum in the database
+    public void resetBossEntries(){
+        try{
+            Connection con = DatabaseConnection.getConnection();
+            PreparedStatement ps = con.prepareStatement("UPDATE boss_entries SET papEntries = ?, zakumEntries = ?, horntailEntries = ?, pinkbeanEntries = ?");
+            for(int i = 1; i < 5; i++){
+                ps.setInt(i, ServerConstants.MAX_DAILY_BOSS_ENTRANCES);
```
Comment (nansonzheng) [inline src/net/server/world/World.java]: By the logic in MapleCharacter.java, entries should be reset to 0.

**Review by nansonzheng — APPROVED** [review]

**Pull request merged**